packages/editor/src/lib/utils/browserCanvasMaxSize.ts

```diff
@@ -1,39 +1,142 @@
-	const maxWidth = await canvasSize.maxWidth({ usePromise: true })
-	const maxHeight = await canvasSize.maxHeight({ usePromise: true })
-	const maxArea = await canvasSize.maxArea({ usePromise: true })
-	return {
-		maxWidth: maxWidth.width,
-		maxHeight: maxHeight.height,
-		maxArea: maxArea.width * maxArea.height,
-	}
-}
+// Extracted from https://github.com/jhildenbiddle/canvas-size
+// MIT License: https://github.com/jhildenbiddle/canvas-size/blob/master/LICENSE
+// Copyright (c) John Hildenbiddle
 
-// https://github.com/jhildenbiddle/canvas-size?tab=readme-ov-file#test-results
 const MAX_SAFE_CANVAS_DIMENSION = 8192
 const MAX_SAFE_CANVAS_AREA = 4096 * 4096
 
+const TEST_SIZES = {
+	area: [
+		// Chrome 70 (Mac, Win)
+		// Chrome 68 (Android 4.4)
+		// Edge 17 (Win)
+		// Safari 7-12 (Mac)
+		16384,
+		// Chrome 68 (Android 7.1-9)
+		14188,
+		// Chrome 68 (Android 5)
+		11402,
+		// Firefox 63 (Mac, Win)
+		11180,
+		// Chrome 68 (Android 6)
+		10836,
+		// IE 9-11 (Win)
+		8192,
+		// IE Mobile (Windows Phone 8.x)
+		// Safari (iOS 9 - 12)
+		4096,
+		// Failed
+		1,
+	],
+	height: [
+		// Safari 7-12 (Mac)
+		// Safari (iOS 9-12)
+		8388607,
+		// Chrome 83 (Mac, Win)
+		65535,
+		// Chrome 70 (Mac, Win)
+		// Chrome 68 (Android 4.4-9)
+		// Firefox 63 (Mac, Win)
+		32767,
+		// Edge 17 (Win)
+		// IE11 (Win)
+		16384,
+		// IE 9-10 (Win)
+		8192,
+		// IE Mobile (Windows Phone 8.x)
+		4096,
+		// Failed
+		1,
+	],
+	width: [
+		// Safari 7-12 (Mac)
+		// Safari (iOS 9-12)
+		4194303,
+		// Chrome 83 (Mac, Win)
+		65535,
+		// Chrome 70 (Mac, Win)
+		// Chrome 68 (Android 4.4-9)
+		// Firefox 63 (Mac, Win)
+		32767,
+		// Edge 17 (Win)
+		// IE11 (Win)
+		16384,
+		// IE 9-10 (Win)
+		8192,
+		// IE Mobile (Windows Phone 8.x)
+		4096,
+		// Failed
+		1,
+	],
+} as const
+
+/**
+ * Tests ability to read pixel data from canvas elements of various dimensions
+ * by decreasing canvas height and/or width until a test succeeds.
+ */
+export function getCanvasSize(dimension: 'width' | 'height' | 'area') {
+	const cropCvs = document.createElement('canvas')
+	cropCvs.width = 1
+	cropCvs.height = 1
+
+	for (const size of TEST_SIZES[dimension]) {
+		const testCvs = document.createElement('canvas')
+		testCvs.width = dimension === 'height' ? 1 : size
+		testCvs.height = dimension === 'width' ? 1 : size
+
+		const cropCtx = cropCvs.getContext('2d')!
+		const testCtx = testCvs.getContext('2d')!
+
+		if (testCtx) {
+			testCtx.fillRect(size - 1, size - 1, 1, 1)
+			cropCtx.drawImage(testCvs, size - 1, size - 1, 1, 1, 0, 0, 1, 1)
+		}
+
+		const isTestPassed = cropCtx && cropCtx.getImageData(0, 0, 1, 1).data[3] !== 0
+
+		// release memory
+		testCvs.width = 0
+		testCvs.height = 0
+
+		if (isTestPassed) {
+			if (dimension === 'area') {
+				return size * size
+			} else {
+				return size
+			}
+		}
+	}
+
+	// release memory
+	cropCvs.width = 0
+	cropCvs.height = 0
+
+	throw Error('Failed to determine maximum canvas dimension')
+}
```

Comment:
No longer a promise as the underlying work is synchronous anyway. Was there a different reason for needing a promise before?